static/app/views/settings/organizationIntegrations/integrationReposAddRepository.tsx

```diff
@@ -35,47 +38,30 @@ export function IntegrationReposAddRepository({
 }: IntegrationReposAddRepositoryProps) {
   const api = useApi({persistInFlight: true});
   const organization = useOrganization();
-  const [dropdownBusy, setDropdownBusy] = useState(false);
   const [adding, setAdding] = useState(false);
-  const [searchResult, setSearchResult] = useState<IntegrationRepoSearchResult>({
-    repos: [],
-  });
-
-  const searchRepositoriesRequest = useCallback(
-    async (searchQuery?: string) => {
+  const [search, setSearch] = useState<string>();
+  const debouncedSearch = useDebouncedValue(search, 200);
+
+  const query = useQuery({
+    queryKey: [
+      `/organizations/${organization.slug}/integrations/${integration.id}/repos/`,
+      {method: 'GET', query: {search: debouncedSearch}},
+    ] as const,
+    queryFn: async context => {
```

Comment:
note that we can’t `useApiQuery` here because it abstracts away the `queryFn`. Since the parent component handles errors and needs to be informed with an `onSearchError` callback, we have to invoke those from the `queryFn`.